tower-http/src/content_encoding.rs

```diff
@@ -201,32 +204,57 @@ pub(crate) fn encodings(
     headers: &http::HeaderMap,
     supported_encoding: impl SupportedEncodings,
 ) -> Vec<(Encoding, QValue)> {
-    headers
+    // Mapping of encodings to corresponding qvalues.
+    let mut encodings = std::collections::BTreeMap::default();
```

Comment:
BTreeMap is probably going to be slower here than a simple vec that you sort after pushing everything to it since the max size here is 4. Even if you also need to check for duplicates before inserting the linear search is most likely going to be faster.

Reply:
I changed the code to store the data in a `Vec<(Encoding, QValue)>`.